blocks/plp/plp.js

```diff
@@ -0,0 +1,528 @@
+import {
+  fetchPlaceholders,
+  readBlockConfig,
+  toClassName,
+  toCamelCase,
+  buildBlock,
+  decorateBlock,
+  loadBlock,
+} from '../../scripts/aem.js';
+
+/**
+ * Fetches and filters products from the product index.
+ * @param {Array<string>|Object} config - An array of product paths or a config object
+ * @param {Object} facets - Optional facets object to calculate counts for filtering options
+ * @returns {Promise<Array<Object>>} Array of filtered product objects
+ */
+export async function lookupProducts(config, facets = {}) {
+  if (!window.productIndex) {
+    // fetch the main product index
+    const resp = await fetch('/us/en_us/products/index.json');
+    const json = await resp.json();
+
+    // build a lookup map of SKU > product data
+    const skuIndex = {};
+    json.data.forEach((row) => {
+      skuIndex[row.sku] = row;
+    });
+
+    // fetch the catalog config
+    const catalog = await fetch('/us/en_us/products/config/catalog.json');
```

Comment:
As mentioned.. I'm going to make sure all the gaps to remove this are in the index.